### ISSUE TITLE
Package Request: stm32cubemx

### ISSUE THREAD

**Issue by ardadem** [void-linux/void-packages#35499]
https://www.st.com/en/development-tools/stm32cubemx.html

> Graphical software configuration tool for STM32 micro controllers that allows generating C initialization code.

**Comment by classabbyamp** [comment]
Yes it costs nothing but it requires agreeing to ToS and giving STmicro your name and email to get the download link. This is probably not very packagable

**Comment by basicfunc** [comment]
> This is probably not very packagable

True

**Comment by ardadem** [comment]
There is a static download link: https://sw-center.st.com/packs/resource/library/stm32cube_mx_v${version}-lin.zip. But i don't know their license allow redistribute.

**Comment by classabbyamp** [comment]
I tried that url with version 6.5.0 and got:
```
<Error>
<Code>NoSuchKey</Code>
<Message>The specified key does not exist.</Message>
<Key>packs/resource/library/stm32cube_mx_v6.5.0-lin.zip</Key>
<RequestId>some string</RequestId>
<HostId>some base64 string</HostId>
</Error>
```

**Comment by classabbyamp** [comment]
nevermind, it has to be the version without `.`s. Taking a look at the [AUR pkgbuild](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=stm32cubemx), this is not a source distribution, but the resulting binary, which decreases the likelihood of acceptance to void-packages.

**Comment by ardadem** [comment]
Yeah, maybe in non-free but still i am even open to only template (like discord). It is necessary to develop HAL applications on STM32.

**Comment by ericonr** [comment]
https://github.com/void-linux/void-packages/pull/22843

IIRC I didn't manage to find anything conclusive in the ToC to say I couldn't redistribute it, but I decided to avoid it for now.

**Comment by ardadem** [comment]
@ericonr Oh, didn't see the pr. Could you consider re-open it?
